src/status_im/contexts/wallet/send/routes/view.cljs

```diff
@@ -89,31 +85,171 @@
+             currency-symbol                            (rf/sub [:profile/currency-symbol])
+             send-from-locked-amounts                   (rf/sub
+                                                         [:wallet/wallet-send-from-locked-amounts])
+             {account-color :color}                     (rf/sub [:wallet/current-viewing-account])
+             locked-amount                              (get send-from-locked-amounts chain-id)
+             network-name-str                           (string/capitalize (name network-name))
+             [input-state set-input-state]              (rn/use-state (cond-> controlled-input/init-state
+                                                                        locked-amount
+                                                                        (controlled-input/set-input-value
+                                                                         locked-amount)))
+             [crypto-currency? set-crypto-currency]     (rn/use-state true)
+             conversion-rate                            (-> token
+                                                            :market-values-per-currency
+                                                            currency
+                                                            :price)
+             {token-balance :total-balance}             (rf/sub [:wallet/token-by-symbol
+                                                                 (str token-symbol)
+                                                                 [chain-id]])
+             current-crypto-limit                       (utils/get-standard-crypto-format
+                                                         token
+                                                         token-balance)
+             current-fiat-limit                         (.toFixed (* token-balance conversion-rate) 2)
+             current-limit                              (if crypto-currency?
+                                                          current-crypto-limit
+                                                          current-fiat-limit)
+             crypto-decimals                            token-decimals
+             input-amount                               (controlled-input/input-value input-state)
+             [is-amount-locked? set-is-amount-locked]   (rn/use-state (some? locked-amount))
+             bottom                                     (safe-area/get-bottom)
+             amount-in-crypto                           (if crypto-currency?
+                                                          input-amount
+                                                          (number/remove-trailing-zeroes
+                                                           (.toFixed (/ input-amount
+                                                                        conversion-rate)
+                                                                     crypto-decimals)))
```

Comment:
I see some duplication in this block similar to the `input_amount` screen. Potentially a place for bugs in the future if changes were not synced. I think we need to make it more reusable, for example common subs and utils can be grouped into a common `init` method, for example: https://github.com/status-im/status-mobile/blob/develop/src/status_im/contexts/chat/messenger/composer/utils.cljs#L187

Reply:
@OmarBasem, yes there is a duplication as I stated in PR description. And I desperately want to remove it and also to refactor `input_amount` because it is terrifying) But I'd be happy to do this after this PR merged, just to have the functionality in place.